plugins/outputs/microsoft_fabric/event_stream.go

```diff
@@ -0,0 +1,191 @@
+			e.log.Tracef("metric: %+v", m)
+			continue
+		}
+
+		// Get the batcher for the chosen partition
+		partition := "<default>"
+		if e.partitionKey != "" {
+			if key, ok := m.GetTag(e.partitionKey); ok {
+				partition = key
+			} else if key, ok := m.GetField(e.partitionKey); ok {
+				if k, ok := key.(string); ok {
+					partition = k
+				}
+			}
+		}
+		batchOptions.PartitionKey = &partition
+		if _, found := batches[partition]; !found {
+			batches[partition], err = e.client.NewEventDataBatch(ctx, &batchOptions)
+			if err != nil {
+				return fmt.Errorf("creating batch for partition %q failed: %w", partition, err)
+			}
+		}
+
+		// Add the event to the partition and send it if the batch is full
+		err = batches[partition].AddEventData(&azeventhubs.EventData{Body: payload}, nil)
+		if err == nil {
+			continue
+		}
+
+		// If the event doesn't fit into the batch anymore, send the batch
+		if !errors.Is(err, azeventhubs.ErrEventDataTooLarge) {
+			return fmt.Errorf("adding metric to batch for partition %q failed: %w", partition, err)
+		}
+
+		// The event is larger than the maximum allowed size so there
+		// is nothing we can do here but have to drop the metric.
+		if batches[partition].NumEvents() == 0 {
+			e.log.Errorf("Metric with %d bytes exceeds the maximum allowed size and must be dropped!", len(payload))
+			e.log.Tracef("metric: %+v", m)
+			continue
+		}
+		if err := e.send(ctx, batches[partition]); err != nil {
+			return fmt.Errorf("sending batch for partition %q failed: %w", partition, err)
+		}
+
+		// Create a new metric and reiterate over the current metric to be
+		// added in the next iteration of the for loop.
+		batches[partition], err = e.client.NewEventDataBatch(ctx, &e.options)
+		if err != nil {
+			return fmt.Errorf("creating batch for partition %q failed: %w", partition, err)
+		}
+		i--
+	}
+
+	// Send the remaining batches that never exceeded the batch size
+	for partition, batch := range batches {
+		if batch.NumBytes() == 0 {
+			continue
+		}
+		if err := e.send(ctx, batch); err != nil {
+			return fmt.Errorf("sending batch for partition %q failed: %w", partition, err)
+		}
+	}
+	return nil
+}
+
+func (e *eventstream) parseconnectionString(cs string) error {
+	// Parse the connection string
+	if cs == "" {
+		return errors.New("connection string must not be empty")
+	}
+	// Split the connection string into key-value pairs
+	pairs := strings.Split(cs, ";")
+	for _, pair := range pairs {
+		// Split each pair into key and value
+		k, v, found := strings.Cut(pair, "=")
+		if !found {
+			return fmt.Errorf("invalid connection string format: %s", pair)
+		}
+		k = strings.ToLower(strings.TrimSpace(k))
+		v = strings.TrimSpace(v)
+		if slices.Contains(confKeys, k) {
+			switch k {
+			case "partitionkey", "partition key":
+				e.partitionKey = v
+			case "maxmessagesize", "max message size":
+				if sz, err := strconv.ParseInt(v, 10, 64); err != nil {
+					e.maxMessageSize = config.Size(sz)
+				}
+			}
+		}
```

Comment:
This will never be the case as you called `strings.ToLower` but the `confKeys` do contain uppercase letters! This is a good example of hard-to-detect-errors due to the definition being artificially move to a variable!

Better:

```suggestion
		switch strings.ReplaceAll(k, " ", "") {
		case "partitionkey":
			e.partitionKey = v
		case "maxmessagesize":
			if sz, err := strconv.ParseInt(v, 10, 64); err != nil {
				e.maxMessageSize = config.Size(sz)
			}
		}
```